doc/code/qml_pauli.rst

```diff
@@ -36,36 +36,35 @@ approach to represent Pauli words. A :class:`~pennylane.pauli.PauliWord` can be
 instantiated by passing a dictionary of wires and their associated Pauli operators.
 
 >>> from pennylane.pauli import PauliWord
->>> pw1 = qml.pauli.PauliWord({0:"X", 1:"Z"})  # X@Z
->>> pw2 = qml.pauli.PauliWord({0:"Y", 1:"Z"})  # Y@Z
+>>> pw1 = PauliWord({0:"X", 1:"Z"})  # X@Z
+>>> pw2 = PauliWord({0:"Y", 1:"Z"})  # Y@Z
 >>> pw1, pw2
 (X(0) @ Z(1), Y(0) @ Z(1))
 
 The purpose of this class is to efficiently compute products of Pauli words and
 obtain the matrix representation.
 
->>> pw1 * pw2
+>>> pw1 @ pw2
```

Comment:
Yeah I agree with all that, I was also a bit confused. I wasnt part of the development of the original class, so can only guess what the reasons were:
- returning the tuple `(op_res, coeff)` is slightly more lightweight than returning a `PauliSentence`
```
def pauli_matmul(pw1, pw2):
    op_res, coeff = pw1 @ pw2
    return PauliSentence({op_res: coeff})
>>> %timeit pw1 @ pw2 
740 ns ± 16.7 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
>>> %timeit pauli_matmul(pw1, pw2)
973 ns ± 19.7 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```
- For some arithmetic you may want to stay with just "pure" pauliwords and tracking coefficients separately is not an issue. This may be because of performance considerations (see point above), or that linear independence between PauliWords is just equality (up to a factor). Think PauliWords are the canonical basis `(1,0,0..)`, `(0, 1, 0..)`, `(0, 0, 1,..)` vs PauliSentence are arbitrary vectors `(a, b, c, ..)`, ... and you sometimes just wanna stay in the canonical basis. That being said I agree that from a user perspective its odd to not get a `PauliSentence`
- I think a lot of focus / motivation for this class in the first place was efficiently creating (sparse) matrices, hence the focus on that in the docs.

I think these are all valid concerns, maybe we move them to a separate PR/forum though?

Reply:
Exactly. The motivation here was "fast", which meant lightweight.  And I honestly didn't even think about it.

I'd be fine with switching to having `@` return a pauli sentence, as long as we have a different method for the more lightweight version that can be used in big problems.

Reply:
Lets move this discussion to this PR https://github.com/PennyLaneAI/pennylane/pull/5054
But essentially this is doing as you propose: Have `@` return a paulisentence but offer a different method for more lightweight processing :-)